openstack/nova-hypervisor-agents/templates/hypervisors-kvm-daemonset.yaml

```diff
@@ -0,0 +1,289 @@
+{{- $hypervisor := merge .Values.defaults.hypervisor.kvm .Values.defaults.hypervisor.common }}
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: {{ .Release.Name }}-compute-kvm
+  labels:
+    system: openstack
+    type: backend
+    component: nova
+spec:
+  revisionHistoryLimit: {{ .Values.pod.lifecycle.upgrades.deployments.revision_history }}
+  selector:
+    matchLabels:
+      name: nova-compute-kvm
+  template:
+    metadata:
+      labels:
+{{ tuple . "nova" "compute" | include "helm-toolkit.snippets.kubernetes_metadata_labels" | indent 8 }}
+        name: nova-compute-kvm
+        alert-tier: os
+        alert-service: nova
+        hypervisor: "kvm"
+      annotations:
+        configmap-etc-hash: {{ include (print .Template.BasePath "/etc-configmap.yaml") . | sha256sum }}
+        configmap-kvm-etc-hash: {{ include "kvm_configmap" . | sha256sum }}
+    spec:
+      terminationGracePeriodSeconds: {{ $hypervisor.default.graceful_shutdown_timeout | default .Values.defaults.default.graceful_shutdown_timeout | add 5 }}
+      hostNetwork: true
+      hostPID: true
+      hostIPC: true
+      securityContext:
+        supplementalGroups: [476, 477]
+        fsGroup: 42436
+      tolerations:
+      {{- range $k, $v :=  merge $hypervisor.pod.tolerations .Values.pod.tolerations }}
+      - # {{ $k }}
+      {{- toYaml $v | nindent 8 }}
+      {{- end }}
+      nodeSelector:
+      {{- range $k, $v := $hypervisor.pod.nodeSelector }}
+        {{ $k }}: {{ $v }}
+      {{- end }}
+      hostAliases:
+      {{- range $ip := .Values.rabbitmq.externalIPs }}
+      - ip: {{ $ip }}
+        hostnames:
+        - {{ default "rabbitmq" $.Values.rabbitmq.nameOverride | printf "nova-%s" | trunc 63 | replace "_" "-" | trimSuffix "-" }}
```

Comment:
Q: Will we deploy one of these charts per AZ/cell?

Reply:
TLDR: Yes.

We would deploy one per cluster. We most likely deploy at least one cluster per AZ, but possibly multiple. There is even the idea to have one of the these per cluster.

But chances are, the charts are gone until then and replaced by an operator.